### PR TITLE
Fix issue #5331

### PR DESCRIPTION
**Pull request by IcaursMoo**
## Fixes
Fixes [#5331](https://github.com/ankidroid/Anki-Android/issues/5331)

## Approach
Update the cards ID everytime the button are selected.

## How Has This Been Tested?

Just to get and compare the card ID.

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [ ] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java [hunk 1]
```diff
@@ -194,6 +197,7 @@
 
     private String[] mSourceText;
 
+    private boolean mTest;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:200]: Codacy Issue found: [Avoid unused private fields such as 'mTest'.](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5609208)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5609208&bid=18378343)

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java [hunk 2]
```diff
@@ -379,6 +379,17 @@ private Bundle getFieldsAsBundle() {
         return fields;
     }
 
+    /**Update Deck ID of current cards.
+     * @param card the card need to update Deck ID.
+     * @Retrun If success, return current Deck ID. Otherwise, return -1.
+     * */
+    private long updateCardCurrentDid(Card card) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java]: You don't need to return anything here, as it's unused.

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java [hunk 3]
```diff
@@ -379,6 +379,17 @@ private Bundle getFieldsAsBundle() {
...
+        if(card == null) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java]:
```suggestion
        if (card == null) {
```

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java [hunk 4]
```diff
@@ -509,6 +520,7 @@ protected void onCollectionLoaded(Collection col) {
             public void onItemSelected(AdapterView<?> parent, View view, int pos, long id) {
                 // Timber.i("NoteEditor:: onItemSelected() fired on mNoteDeckSpinner with pos = "+Integer.toString(pos));
                 mCurrentDid = mAllDeckIds.get(pos);
+		    updateCardCurrentDid(mCurrentEditedCard);
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java]:
```suggestion
		  updateCardCurrentDid(mCurrentEditedCard);
```

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5609208&bid=18378343)

**Comment by david-allison-1** [comment]
Another commit came in while I was reviewing, please `@` me if I don't get around to reviewing this within 12 hours.

**Comment by IcaursMoo** [comment]
Actually, you can ignore the first commit. For some resonses, I need to get 2 pull requests.The later commit fix the both #5331 and #6253.&nbsp;
I have test some simple case, like modify the deck ID and Type at the same time.&nbsp;
One thing is that it does not work for additionally added text since it not allowed user to modify the contents.
&nbsp;









莫宇
南方科技大学


 联系我









&nbsp;
&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: "david-allison-1"; 
Date: 2020年5月30日(星期六) 凌晨1:24
To: "ankidroid/Anki-Android"; 
Cc: "IcaursMoo"<11710815@mail.sustech.edu.cn>; "Author"; 
Subject: Re: [ankidroid/Anki-Android] Fix issue #5331 (#6299)

&nbsp;


 
Another commit came in while I was reviewing, please @ me if I don't get around to reviewing this within 12 hours.

**Comment by IcaursMoo** [comment]
Sorry, I forget to add the version. This is work for 2.10alpha3. So I just add a few lines of code based on the released version&nbsp;2.10alpha3.











莫宇
南方科技大学


 联系我









&nbsp;
&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java [hunk 5]
```diff
@@ -286,40 +288,6 @@ public void onPostExecute(CollectionTask.TaskData result) {
         }
     };
 
-    private void displayErrorSavingNote() {
-        int errorMessageId = getAddNoteErrorResource();
-        UIUtils.showThemedToast(this, getResources().getString(errorMessageId), false);
-    }
-
-
-    protected @StringRes int getAddNoteErrorResource() {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:295]: This code is still removed in the latest commit

**Comment by IcaursMoo** [comment]
This piece of codes are irrelavante to #5331 or #6253 in my tests.









莫宇
南方科技大学


 联系我









&nbsp;
&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------

**Comment by david-allison-1** [comment]
> This piece of codes are irrelavante to #5331 or #6253 in my tests.

When this pull request is merged, it becomes the new "latest version" of AnkiDroid. 

If this removes code, or reverts bugfixes, then our users will be unhappy. 

Reading through reverted code also makes it harder to understand the change that you made.

It'll need to be fixed before the pull request is merged. Please let me know if I can help.

**Pull request closed without merging**